From 4a3fbfcc343728a93ca4c9d004a1ce4626782065 Mon Sep 17 00:00:00 2001
From: Pavel Grunt <pgrunt@redhat.com>
Date: Fri, 20 Mar 2015 10:35:55 +0100
Subject: [PATCH] Clear GError in cleanup section

(cherry picked from commit 526e882368eb74eb1ed9de5c1c8aefb7b428fb9d)
---
 src/remote-viewer-main.c | 3 +--
 src/virt-viewer-main.c   | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/remote-viewer-main.c b/src/remote-viewer-main.c
index b689033..8298053 100644
--- a/src/remote-viewer-main.c
+++ b/src/remote-viewer-main.c
@@ -138,7 +138,6 @@ main(int argc, char **argv)
         g_printerr(_("%s\nRun '%s --help' to see a full list of available command line options\n"),
                    error->message, base_name);
         g_free(base_name);
-        g_error_free(error);
         goto cleanup;
     }
 
@@ -181,7 +180,6 @@ main(int argc, char **argv)
     if (!virt_viewer_app_start(app, &error)) {
         if (g_error_matches(error, VIRT_VIEWER_ERROR, VIRT_VIEWER_ERROR_CANCELLED))
             ret = 0;
-        g_clear_error(&error);
         goto cleanup;
     }
 
@@ -197,6 +195,7 @@ main(int argc, char **argv)
     if (viewer)
         g_object_unref(viewer);
     g_strfreev(args);
+    g_clear_error(&error);
 
     return ret;
 }
diff --git a/src/virt-viewer-main.c b/src/virt-viewer-main.c
index a55b4f1..713cdec 100644
--- a/src/virt-viewer-main.c
+++ b/src/virt-viewer-main.c
@@ -98,7 +98,6 @@ int main(int argc, char **argv)
     if (error) {
         g_printerr("%s\n%s\n",
                    error->message, help_msg);
-        g_error_free(error);
         goto cleanup;
     }
 
@@ -121,7 +120,6 @@ int main(int argc, char **argv)
     if (!virt_viewer_app_start(VIRT_VIEWER_APP(viewer), &error)) {
         if (g_error_matches(error, VIRT_VIEWER_ERROR, VIRT_VIEWER_ERROR_CANCELLED))
             ret = 0;
-        g_clear_error(&error);
         goto cleanup;
     }
 
@@ -135,6 +133,7 @@ int main(int argc, char **argv)
     g_free(uri);
     g_strfreev(args);
     g_free(help_msg);
+    g_clear_error(&error);
 
     return ret;
 }
